openforcefield/topology/molecule.py

```diff
@@ -1788,7 +1786,9 @@ def __eq__(self, other):
            No effort is made to ensure that the atoms are in the same order or that any annotated properties are preserved.
 
         """
-        return self.is_isomorphic(other)
+        # updated to use the new isomorphic checking method, with full matching
+        # TODO the doc string did not match the previous function what matching should this method do?
+        return Molecule.are_isomorphic(self, other, return_atom_map=False)[0]
```

Comment:
@j-wags I have changed this method to use the new full isomorphic matching method, but is this what we want? The docstring suggests we are checking the smiles and I think that's whats the old `is_isomorphic` function used to do, do we want to go back to that or keep this more strict matching checking?

Reply:
Ah, I like what you've done here.

_Just in case_ the default kwargs for `are_isomorphic` change in the future, and to make this function more readable, could you explicitly set the kwargs to `are_isomorphic` here (so, like, everything is `True`, except checking for identical atom ordering)?